packages/@aws-cdk/aws-stepfunctions/README.md

```diff
@@ -258,6 +259,81 @@ const map = new stepfunctions.Map(this, 'Map State', {
 map.iterator(new stepfunctions.Pass(this, 'Pass State'));
 ```
 
+### Custom
+
+It's possible that the high-level constructs for the states or `stepfunctions-tasks` do not have
+the states or service integrations you are looking for. The primary reasons for this lack of
+functionality are:
+
+* A [service integration](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-service-integrations.html) is available through Amazon States Langauge, but there are no Construct
+  classes for them
+* The state or state properties are available through Step Functions, but are not configurable
+  through constructs
+
+If a feature is not available, a `CustomState` can be used to supply any Amazon States Language
+JSON-based object as the state definition.
+
+[Code Snippets](https://docs.aws.amazon.com/step-functions/latest/dg/tutorial-code-snippet.html#tutorial-code-snippet-1) are available and can be plugged in as the state definition.
+
+All of the state properties will be embedded in the JSON that defines it. Custom states can be
+chained together and used like any other state. The `toStateJson` API returns the state JSON
```

Comment:
toStateJson is a private API between the State Machine and the State. Don't advertise it.

Reply:
gotcha, should we be marking it `@internal` or something similar?

Reply:
Too late for that.